static/app/utils/profiling/flamegraph/flamegraphStateProvider/flamegraphSearch.tsx

```diff
@@ -23,10 +20,16 @@ type FlamegraphSearchArrowNavigationAction = {
   type: 'set search index position';
 };
 
+type SetSearchQuery = {
+  payload: string;
+  type: 'set search query';
+};
+
 type FlamegraphSearchAction =
   | ClearFlamegraphSearchAction
   | FlamegraphSearchArrowNavigationAction
-  | SetFlamegraphResultsAction;
+  | SetFlamegraphResultsAction
+  | SetSearchQuery;
```

Comment:
How does this affect the undo/redo functionalities? Can undo result in a state where the search string is set but the search results aren't?

Reply:
No, but this is misleading. Since the query will change, the effect in FlamegraphSearch will trigger and the results will be set again. Idk what would be a nicer way to do this